docs/lang/articles/misc/profiler.md

```diff
@@ -4,78 +4,125 @@ sidebar_position: 4
+    - In `'count'` mode (default), records with the same kernel name are counted as a profiling result,
+    and then presented in a statistical perspective.
+    - In `'trace'` mode, the profiler shows you a list of kernels that were launched on hardware during the profiling period.
+    This mode provides more detailed performance information and runtime hardware metrics for each kernel.
+3. To clear records in this profiler, call `ti.clear_kernel_profile_info()`.
 
+For example:
 
-@ti.kernel
-def compute():
-    var[0] = 1.0
-    print("Setting var[0] =", var[0])
+```python {3,13}
+import taichi as ti
 
+ti.init(ti.cpu, kernel_profiler=True)
+x = ti.field(ti.f32, shape=1024*1024)
 
-compute()
-ti.print_profile_info()
+@ti.kernel
+def fill():
+    for i in x:
+        x[i] = i
+
+for i in range(8):
+    fill()
+ti.print_kernel_profile_info('trace')
+ti.clear_kernel_profile_info() # clear all records
+
+for i in range(100):
+    fill()
+ti.print_kernel_profile_info() # default mode: 'count'
 ```
 
-`ti.print_profile_info()` prints profiling results in a hierarchical format.
+The outputs would be:
+
+```
+=========================================================================
+X64 Profiler(trace)
+=========================================================================
+[      % |     time    ] Kernel name
+[  0.00% |    0.000  ms] jit_evaluator_0_kernel_0_serial
+[ 60.11% |    2.668  ms] fill_c4_0_kernel_1_range_for
+[  6.06% |    0.269  ms] fill_c4_0_kernel_1_range_for
+[  5.73% |    0.254  ms] fill_c4_0_kernel_1_range_for
+[  5.68% |    0.252  ms] fill_c4_0_kernel_1_range_for
+[  5.61% |    0.249  ms] fill_c4_0_kernel_1_range_for
+[  5.63% |    0.250  ms] fill_c4_0_kernel_1_range_for
+[  5.61% |    0.249  ms] fill_c4_0_kernel_1_range_for
+[  5.59% |    0.248  ms] fill_c4_0_kernel_1_range_for
+-------------------------------------------------------------------------
+[100.00%] Total kernel execution time:   0.004 s   number of records:  9
+=========================================================================
+=========================================================================
+X64 Profiler(count)
+=========================================================================
+[      %     total   count |      min       avg       max   ] Kernel name
+[100.00%   0.033 s    100x |    0.244     0.329     2.970 ms] fill_c4_0_kernel_1_range_for
+-------------------------------------------------------------------------
+[100.00%] Total kernel execution time:   0.033 s   number of records:  1
+=========================================================================
+```
 
 :::note
-`ScopedProfiler` is a C++ class in the core of Taichi. It is not exposed
-to Python users.
+Currently the result of `KernelProfiler` could be incorrect on OpenGL backend due to its lack of support for `ti.sync()`.
 :::
+
+### Advanced mode
+
+For the CUDA backend, `KernelProfiler` has an experimental GPU profiling toolkit, Nvidia CUPTI, which provides low and
```

Comment:
...has an experimental GPU profiling toolkit, which is based on the Nvidia CUPTI that has low and deterministic profiling overhead, and is able...